test/integ_tests/mock_utils.py

```diff
@@ -153,8 +158,11 @@ class Boto3SessionAllWrapper(SessionWrapper):
     def __init__(self):
         super().__init__()
         boto3.Session = self
+        self._region = None
 
     def __call__(self, *args, **kwargs):
+        if "region_name" in kwargs:
+            self._region = kwargs["region_name"]
```

Comment:
It shouldn't matter, but none of it should matter (since it's all offline anyway), and it does matter because we do these extra checks. Issues that are caused by side effects like this can be really hard to debug, so maybe if there's a clean way to fix it we should.

Reply:
I agree, fixed